src/signTxMint.c

```diff
@@ -33,7 +33,7 @@ static inline void advanceState()
 		ASSERT(subctx->currentAssetGroup < subctx->numAssetGroups);
 
 		// we are going to receive token amounts for this group
-		ASSERT(subctx->numTokens > 0);
+		VALIDATE(subctx->numTokens > 0, ERR_INVALID_DATA);
```

Comment:
The value of `subctx->numTokens` should be validated immediately after it is received (as we do with other values). The assertion is all right here, ensures that we have not messed up something before (and we did, by forgetting the validation).

Reply:
See above.